client/blocks/app-banner/index.jsx

```diff
@@ -58,6 +58,19 @@ export class AppBanner extends Component {
 		userAgent: typeof window !== 'undefined' ? window.navigator.userAgent : '',
 	};
 
+	constructor( props ) {
+		super( props );
+
+		if (
+			typeof window !== 'undefined' &&
+			window.sessionStorage.getItem( 'wpcom_signup_complete_show_draft_post_modal' )
+		) {
+			this.state = { isDraftPostModalShown: true };
+		} else {
+			this.state = { isDraftPostModalShown: false };
+		}
+	}
```

Comment:
Do we really need to read `wpcom_signup_complete_show_draft_post_modal` here at all? It seems to me that the entire job is already done on the `/post` page, when computing the `showDraftPostModal` value. There it will disable the `AppBanner` while the draft post modal is being shown.

And because the `showDraftPostModal()` function is always called before the `AppBanner` is rendered, and because it deletes the `wpcom_signup_complete_show_draft_post_modal` key, the value read in this constructor will always be `false`.

Reply:
I read `wpcom_signup_complete_show_draft_post_modal` here because, in the tests I've done, the App Banner constructor always gets called before the `/post` page, so it succeeds in retrieving the session storage value. But I may be missing something and be testing it all wrongly 🤔 

`showDraftPostModal()` value, that is called after the App Banner, at the moment is only passed through `CalypsoifyIframe` and used in the Welcome Tour (in ETK) to show the Draft Post Modal or the Welcome Tour. App Banner computation happens before! 

It would be great if we could somehow share the value between `controller.js` and `app-banner.js`, having the check only in one part of the application

Reply:
> the App Banner constructor always gets called before the `/post` page

Oh yes now I see how this can happen: the `AppBanner` is rendered in the root `Layout` component, which is [rendered as a "boot placeholder"](https://github.com/Automattic/wp-calypso/blob/trunk/client/boot/common.js#L436) before any route handler runs. That means you need to read the session storage flag in `AppBanner` constructor, too.

Reply:
One more thing: after the `this.state.isDraftPostModalShown` is assigned in the constructor, it never changes, and if it's `true`, the banner is disabled forever, until the next full page reload. Not sure if that's OK.

Reply:
Valid observation. From the [original PR](https://github.com/Automattic/wp-calypso/pull/58524):
> We also want to show the Mobile app banner only at the next visit after the blogger modal has been displayed: pdgK6S-ku-p2

I think that is to not overload the user with modals and banners, so if a user gets to see the Draft Post Modal he will see the App Banner on the next visit/refresh. We may ask @simison for a confirmation on this, just to be sure! 😄

Reply:
Showing the mobile modal on "next visits" was suggested in pdgK6S-ku-p2 but I don't think it turned into a decision in that convo. I do see it's how it was already implemented in https://github.com/Automattic/wp-calypso/pull/58524 and I might be missing some other convo elsewhere:

> blogger modal (if the user is in the Blogger flow)
> app banner (at the next visit to the Editor)
> editor welcome tour (immediately after dismissing the app banner)

To me, it still makes sense to show mobile banner first, but I will defer to our mobile web designer @annchichi if what was implemented is good, or if there are needs for changes.

Lemme know if you need help deciding.

Reply:
Thank you @simison! Ok, let's wait for @annchichi to take a look 👍

Reply:
Hey hey @escapemanuele, I am so sorry I am missing out on this! 

- The implementation works as Yael (the convo) that the banner placed before the tour. ([link](p1642541404027600/1634738497.202200-slack-C029WSAQMT9))
- I noticed the copy "No, thanks" can be changed to "Not now" to evaluate the timing of this banner. 
- One more thing, the tour is quite hard to perform swipe gestures on the left and right. The users will need to tap the little dot pagination to get to the tour's next tooltips. It would be great if we could improve that. I am not sure if this is the right place. I am happy to take a comment on where it should be. 

Hope this helps! :D

Reply:
Thanks @annchichi for your insight! 🙏 

I'm going to deploy this PR as it solves some performance issues, I asked @annchichi to add a comment about the other improvements (mobile swipe gestures) for the tour kit in [another issue](https://github.com/Automattic/wp-calypso/issues/60537)